rotkehlchen/tests/api/test_evm_transactions.py

```diff
@@ -89,3 +125,111 @@ def test_query_transactions(rotkehlchen_api_server: 'APIServer'):
         for entry in result['entries']:
             assert entry['entry']['timestamp'] <= last_ts
             last_ts = entry['entry']['timestamp']
+
+
+@pytest.mark.parametrize('ethereum_accounts', [[ADDY]])
+@pytest.mark.parametrize('is_async_query', [True, False])
+def test_evm_transaction_hash_addition(rotkehlchen_api_server: 'APIServer', is_async_query: bool) -> None:  # noqa: E501
+    tx_hash = '0xf7049668cb7cbb9c00d80092b2dce7ea59984f4c52c83e5c0940535a93f3d5a0'
+    chain_id = ChainID.ETHEREUM
+    expected_decoded_events = [
+        # gas transaction is only added for tracked accounts.
+        HistoryBaseEntry(
+            identifier=1,
+            event_identifier=HistoryBaseEntry.deserialize_event_identifier(tx_hash),
+            sequence_index=22,
+            timestamp=TimestampMS(1513958719000),
+            location=Location.ETHEREUM,
+            event_type=HistoryEventType.INFORMATIONAL,
+            event_subtype=HistoryEventSubType.NONE,
+            asset=A_ETH,
+            balance=Balance(),
+            location_label='0x01349510117dC9081937794939552463F5616dfb',
+            notes='Create CDP 131',
+            counterparty=CPT_SAI,
+        ),
+    ]
+    # first check that there's no such transaction in the db
+    with rotkehlchen_api_server.rest_api.rotkehlchen.data.db.conn.read_ctx() as cursor:
```

Comment:
Just a small nitpick but you can do

```suggestion 
    database = rotkehlchen_api_server.rest_api.rotkehlchen.data.db
    with database.conn.read_ctx() as cursor:
```

and the same in the other places in this test